storage: log when rebalance-after-split causes snapshot

tbg opened the pull request:
This will help track down (by means of log inspection) how big a deal
this is.

Release note: None

tbg committed:
storage: log when rebalance-after-split causes snapshot

This will help track down (by means of log inspection) how big a deal
this is.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/32549)
<!-- Reviewable:end -->

tbg committed:
storage: log info when gc'ing recent preemptive snap

GC'ing a preemptive snapshot is necessary if the corresponding
replication change failed, but if it didn't, it causes a Raft snapshot.
Unfortunately, it's impossible to decide which one is which based on
the local information. Even worse, an obsolete preemptive snapshot can
also block required overlapping snapshots, so simply delaying the GC
is also a dangerous option. For now, log at INFO (in principle these
events can already be grepped from the logs, but I expect the
explicitness to be useful still).

Release note: None

tbg commented:
bors r=petermattis
